src/bentoml/_internal/frameworks/transformers.py

```diff
@@ -750,10 +751,29 @@ def import_model(
         resume_download=resume_download,
         **extra_hf_hub_kwargs,
     )
+
+    is_auto_class = False
     if pretrained_model_class is None:
-        pretrained_model_class = getattr(transformers, config.architectures[0])
-        logger.info(
-            f"pretrained_model_class is not provided, bentoml will create a model with the following pretrained model class {config.architectures[0]}. Available pretrained classes for this model: {config.architectures}."
+        if getattr(config, "architectures", None):
+            try:
+                pretrained_model_class = getattr(transformers, config.architectures[0])
```

Comment:
You shouldn't do this eagerly, I think having the string is fine. Transformers has a mapping maintaining all of the supported architecture (Either CONFIG_MAPPING or AUTO_MODEL_MAPPING or sth)

Then it won't break with trust_remote_code. We can't really rely on the models on the Hub to always contain the `auto_map`

Reply:
I can only see two ways for us to automatically determine the model class for the user (config.architectures or config.auto_map). So if a repository does not have `architectures` AND `auto_map`, then `import_model` can't determine the model class.

Reply:
What happens if `architectures=[]`?This will fail

Reply:
Yeah if architectures=[], then will try to look at auto_map. If both empty, then there's nothing more we could do 😞

Reply:
> You shouldn't do this eagerly, I think having the string is fine. Transformers has a mapping maintaining all of the supported architecture (Either CONFIG_MAPPING or AUTO_MODEL_MAPPING or sth)

I assume what you wanted was for me to use the `MODEL_MAPPING` and check if `config.architectures[0]` was in the map. But after playing around with `MODEL_MAPPING` and `CONFIG_MAPPING`, it appears that they are quite expensive to use. They are lazy loading maps that, when initialized, will reference a lot of classes in transformers. When I referenced the map, it took about a minute for the map to initialize. 

Below is a sample of the python output of `MODEL_MAPPING`: 
```
[(<class 'transformers.models.albert.configuration_albert.AlbertConfig'>, <class 'transformers.models.albert.modeling_albert.AlbertModel'>),
(<class 'transformers.models.align.configuration_align.AlignConfig'>, <class 'transformers.models.align.modeling_align.AlignModel'>),
```

Reply:
They have the name variant, we don't need to use the lazy map, so `MODEL_MAPPING_NAMES` and `CONFIG_MAPPING_NAMES`

Reply:
I've also taken a look into the name variants, but they don't seem to work for our use case. 
`MODEL_NAMES_MAPPING` has something like this:
```
("albert", "ALBERT"),
        ("align", "ALIGN"),
        ("altclip", "AltCLIP"),
```
But `config.architectures` contains a string that is the name of a model class (eg. T5ForConditionalGeneration). I've found another map that has something similar to what we need but it has the names of the base classes only (eg. T5Model). 

Overall, grabbing the attribute from `transformers` inside of a try block might be the most robust approach.

Reply:
We want to use `CONFIG_MAPPING_NAMES` to get the config class, then use `from_config` right?

Reply:
calling `from_config` will still require the model class, not the config class. The model class is required for calling both `from_pretrained` and `from_config`.

Reply:
Doesn't AutoModel has a from_config option?

Reply:
Yes, AutoModel and equivalents (eg. AutoModelForCasualLM, AutoModelForSeq2Seq, etc) all have a from_config option. But I don't think CONFIG_MAPPING_NAMES and MODEL_NAMES_MAPPING can assist us in this case.

Reply:
I think you can get MODEL_MAPPING[config_class], since __getitem__ is lazy see https://github.com/huggingface/transformers/blob/9beb2737d758160e845b66742a0c01201e38007f/src/transformers/models/auto/auto_factory.py#L734